src/generator/state-machine-to-intermediate.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Converts a variable name to it's intermediate form.
+ * Letter variables such as 'X' are converted to signals
+ * while intermediate values (e.g. 'INT_A') are left untouched
+ * @param {*} varName 
+ */
+const getSymbolFor = varName => {
+    // TODO: Handle chemical signals
+    if(varName.length === 1) {
+        return `signal_${varName}`;
+    } else {
+        return varName;
+    }
+}
+
+/**
+ * Gets the intermediate form of a statement by converting it
+ * into multiple instructions and guarding outputs
+ * @param {*} node - A statement or expression 
+ * @param {boolean} [isRight] - true if the expression is the right operand of an expression
+ */
+const getIntermediateFormOf = ({left, right, operator, out}, isRight) => {
```

Comment:
That's correct. It's a new ES6 feature called destructuring. You can also do it in a function body:

`const {left, right, operator, out} = expression`